### PR TITLE
Upload invoice endpoint

### PR DESCRIPTION
**Pull request by AnthonyBuisset**
- Implement PostgresProjectRewardV2Adapter.findBudgets (#347)
- Fix NEP on KYB
- Update job schedulers thread pool size to 3
- Fix IT
- Invoice preview and list endpoints (#360)
- upload invoice endpoint
- distinguish invoice id from name
- postgres invoice storage
- rename invoice preview storage to invoice storage
- save invoice preview as draft
- delete drafts when creating a new preview
- upload invoice domain logic
- use PdfStoragePort
- list invoices domain logic
- postgres invoice adapter
- make integration test pass with whole flow
- mark invoice as processing when uploaded
- merge invoice and invoice preview
- persist invoice data
- handle sequence number
- add todo

### CODE REVIEW

File: accounting-domain/src/main/java/onlydust/com/marketplace/accounting/domain/model/Invoice.java [hunk 1]
```diff
@@ -0,0 +1,172 @@
...
+                now,
+                now.plusDays(DUE_DAY_COUNT_AFTER_CREATION),
+                Name.of(sequenceNumber, companyInfo.name),
+                Status.DRAFT,
+                companyInfo.vatRegulationState() == Invoice.VatRegulationState.APPLICABLE ? BigDecimal.valueOf(0.2) : BigDecimal.ZERO
+        ).companyInfo(companyInfo);
+    }
+
+    public enum Status {
+        DRAFT, PROCESSING, REJECTED, APPROVED;
```
Comment (PierreOucif) [inline accounting-domain/src/main/java/onlydust/com/marketplace/accounting/domain/model/Invoice.java:69]: `COMPLETE` le status de mémoire, à checker dans le ticket.

File: accounting-domain/src/main/java/onlydust/com/marketplace/accounting/domain/service/AccountingService.java [hunk 2]
```diff
@@ -167,6 +179,16 @@ public List<PayableReward> getPayableRewards() {
                 .toList();
     }
 
+    private <ID> void onAllowanceUpdated(ID id, Currency currency, AccountBookState accountBook) {
```
Comment (PierreOucif) [inline accounting-domain/src/main/java/onlydust/com/marketplace/accounting/domain/service/AccountingService.java:182]: C'est quoi le rapport avec l'invoicing ?

**Review by PierreOucif — APPROVED** [review]

File: accounting-domain/src/main/java/onlydust/com/marketplace/accounting/domain/service/AccountingService.java (hunk 2 above)
Comment (AnthonyBuisset) [reply accounting-domain/src/main/java/onlydust/com/marketplace/accounting/domain/service/AccountingService.java:182]: Aucun, ca ressemble a un soucis de rebase, je check

File: accounting-domain/src/main/java/onlydust/com/marketplace/accounting/domain/model/Invoice.java (hunk 1 above)
Comment (AnthonyBuisset) [reply accounting-domain/src/main/java/onlydust/com/marketplace/accounting/domain/model/Invoice.java:69]: bizarre pour un statut d'invoice, pour le paiement je comprendrais... je vais verifier

**Pull request merged**